edge-modules/api-proxy-module/src/monitors/config_monitor.rs

```diff
@@ -56,51 +65,66 @@ pub fn start(
     //Special handling of some of the environment variables
     specific_handling_env_var();
 
-    //Parse default config and notify to reboot nginx if it has already started
-    //If the config is incorrect, return error because otherwise nginx doesn't have any config.
-
-    match parse_config() {
-        //Notify watchdog config is there
-        Ok(()) => notify_received_config.notify(),
-        Err(err) => {
-            return Err(anyhow::anyhow!(
-                "Error while parsing default config: {}",
-                err
-            ))
-        }
-    };
+    info!("Starting config monitoring loop");
 
     let monitor_loop: JoinHandle<Result<()>> = tokio::spawn(async move {
         loop {
             let wait_shutdown = shutdown_signal.notified();
             futures::pin_mut!(wait_shutdown);
-
-            let message = match futures::future::select(wait_shutdown, client.next()).await {
-                Either::Left(_) => {
-                    warn!("Shutting down config monitor!");
-                    return Ok(());
-                }
-                Either::Right((Some(Ok(message)), _)) => message,
-                Either::Right((Some(Err(err)), _)) => {
-                    error!("Error receiving a message! {}", err);
-                    continue;
-                }
-                Either::Right((None, _)) => {
-                    warn!("Shutting down config monitor!");
-                    return Ok(());
-                }
-            };
-
-            if let azure_iot_mqtt::module::Message::TwinPatch(twin) = message {
-                if let Err(err) = save_raw_config(&twin) {
-                    error!("received message {}", err);
-                } else {
-                    match parse_config() {
-                        //Notify watchdog config is there
-                        Ok(()) => notify_received_config.notify(),
-                        Err(error) => error!("Error while parsing default config: {}", error),
-                    };
-                }
+            let get_new_sas_token = token_client.get_new_sas_token(Utc::now());
+            futures::pin_mut!(get_new_sas_token);
+            let reload_config = futures::future::select(get_new_sas_token, client.next());
+
+            let parse_config_request =
+                match futures::future::select(wait_shutdown, reload_config).await {
+                    Either::Left(_) => {
+                        warn!("Shutting down config monitor!");
+                        return Ok(());
+                    }
+                    Either::Right((reload_config, _)) => match reload_config {
+                        Either::Left((Ok(Some(token)), _)) => {
+                            info!("New SAS token received, reloading the config");
+                            std::env::set_var("SAS_TOKEN", token);
```

Comment:
Emm, i thought you were going to write this to a file?

Reply:
Env var is visible from outside the container as well.. (docker inspect..) so not sure if this is better.

Reply:
We write it into a file. That env var is not accessible upon docker inspect. I'll check but it should be local to the process

Reply:
1. I pass it as an env VAR because currently API proxy is meant to be configured by user through environment variable. User set env var, API proxy parse the default config and replace place holder with user env var. I set env var here, instead of passing a parameter to keep the previous logic and avoid a refactoring of the function.
2. Now that we have 2 means of configuring that API proxy, will need to refactor that config.

Reply:
Okay, can you pls add a comment to explain 
1. Why you are setting the env var
2. TODO to refactor this in the future

Reply:
Explained above.